bundles/org.openhab.binding.sncf/src/main/java/org/openhab/binding/sncf/internal/handler/StationHandler.java

```diff
@@ -0,0 +1,223 @@
+                return;
+            }
+            freeRefreshJob();
+        }
+        refreshJob = scheduler.schedule(this::queryApiAndUpdateChannels, wishedDelay, TimeUnit.SECONDS);
+    }
+
+    private void queryApiAndUpdateChannels() {
+        SncfBridgeHandler bridgeHandler = getBridgeHandler();
+        if (bridgeHandler != null) {
+            try {
+                bridgeHandler.getNextPassage(stationId, GROUP_DEPARTURE).ifPresentOrElse(departure -> {
+                    getThing().getChannels().stream().map(Channel::getUID).filter(
+                            channelUID -> isLinked(channelUID) && GROUP_DEPARTURE.equals(channelUID.getGroupId()))
+                            .forEach(channelUID -> {
+                                State state = getValue(channelUID.getIdWithoutGroup(), departure, GROUP_DEPARTURE);
+                                updateState(channelUID, state);
+                            });
+                    scheduleRefresh(fromDTO(departure.stopDateTime.departureDateTime));
+                }, () -> {
+                    logger.debug("No {} available", GROUP_DEPARTURE);
+                    scheduleRefresh(ZonedDateTime.now().plusMinutes(5));
+                });
+                bridgeHandler.getNextPassage(stationId, GROUP_ARRIVAL).ifPresentOrElse(arrival -> {
+                    getThing().getChannels().stream().map(Channel::getUID)
+                            .filter(channelUID -> isLinked(channelUID) && GROUP_ARRIVAL.equals(channelUID.getGroupId()))
+                            .forEach(channelUID -> {
+                                State state = getValue(channelUID.getIdWithoutGroup(), arrival, GROUP_ARRIVAL);
+                                updateState(channelUID, state);
+                            });
+                    scheduleRefresh(fromDTO(arrival.stopDateTime.arrivalDateTime));
+                }, () -> {
+                    logger.debug("No {} available", GROUP_ARRIVAL);
+                    scheduleRefresh(ZonedDateTime.now().plusMinutes(5));
+                });
+                updateStatus(ThingStatus.ONLINE);
+            } catch (SncfException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                freeRefreshJob();
+            }
+        }
+    }
+
+    private State getValue(String channelId, Passage passage, String groupDeparture) {
+        switch (channelId) {
+            case DIRECTION:
+                return new StringType(passage.route.direction.name);
+            case CODE:
+                return new StringType(passage.displayInformations.code);
+            case COMMERCIAL_MODE:
+                return new StringType(passage.displayInformations.commercialMode);
+            case NAME:
+                return new StringType(passage.displayInformations.name);
+            case NETWORK:
+                return new StringType(passage.displayInformations.network);
```

Comment:
Maybe you should consider the case the string is null ?

Reply:
Done.